### ISSUE TITLE
No Unused Variables in setup() return

### ISSUE THREAD

**Issue by srz1992** [vuejs/eslint-plugin-vue#1923]
**Please describe what the rule should do:**
When a variable is returned in setup(), but not used in the template, eslint should mark it as an error.

**What category should the rule belong to?**

[ x ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```GOOD
<template>
  <div>{{ usedVariable }}</div>
</template>

<script>
export default {
  name: "App",
  setup() {
    const usedVariable = 'abc'
    const unusedVariable = '123'
    return {
      usedVariable,
    }
  }
};
</script>

```

``` BAD
<template>
  <div>{{ usedVariable }}</div>
</template>

<script>
export default {
  name: "App",
  setup() {
    const usedVariable = 'abc'
    const unusedVariable = '123'
    return {
      usedVariable,
      unusedVariable // this should be an error
    }
  }
};
</script>

```

**Additional context**
n/a

**Comment by ota-meshi** [comment]
I think that you can use the "vue/no-unused-properties" rule. [DEMO]

https://eslint.vuejs.org/rules/no-unused-properties.html

[DEMO]: https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJx1VMty2zAM/BWOLmkz5nia3tw0M/2JXqocIAqSmFCkSpB+1ON/L0hZkuWmF8tcgEtg8TgXytVY7IrngP1gIOBLaYV4rvX+5XwWkbD+CV5DZVBcLs/bhJf2eXvjzUdSXg+B/28fBZLRNiSSsthH3Fono008cvBuQB80UlnsxK+yQO+dL4uNOLNv610cJktypWQpC8IQh7J4FZfX0j5uS4vHwfkgamwgmiDO6SkLPe7Y+cfArpuE5GufPo9mIZSzFNbpfBcPUKmHld3ee3x5+nr18Ezo7cQnVlybGVwTbLcidJoEdS6aWlQowIox63zhkj78c/mWNL2qWGwKHw1SsTtnAZXre7RB1tqjCnrP1XraZMsbHSW/R3IPnjKa0/BRBeclRVY7ow2frredzUiLIaCXY04Z4TIBnaxKVeo1ocQjqsg8k1UBgzUqAx4SD80GA0R8mXQ7U3HMA3iUFlv5B71bcFv/48ohf4QB55z89Rz1aAneGY68xeME1ljFtkW/nA0GTKrMCGshwbe0AsbIe+wr9GsLGk5WNyvsHU+3TkYr7v8sy4TyTITT6iBVx3KppHV+bG0cIFVhzg2PdzLgMXiQlXMGwaaXwtpC2OsJacCY0PEMtd0MRS7nmrE1rgJzB+o+DdQ9aC0H/VG9td2D0fVYgmFGPZ8jO8tDpwPSAGrWxTiW2TXcWqg03VSTG41Tq7Vt/ydUr4+8OTIb9xg3T4BqNlo8SDr1lTMz4qxTgTPkwHXPXyQFwxyIq95YRmNmhuw8HbjzgwsnLnUVtQl6SdnjKAQuAOd+DWvCCE1zJyJ9NGbUQe0ODPKkas64lml7LTQ8OJSa1cPSb2mLyAp5kPFmrvNq5T24HLggmCl7Xo2aFzEuNo/AIvNaWiCCBmWjLUtyukN5cuF28K6wGxLIwjKVtly4xZ53vIEKF3mv4LyeEsQyIbdDBerdY4Me7dIok5EnS3X34LqUBx1GD4+/I+82edJo6ozENLtkYYx9bNZUVpeEulz+AuozXP0=

**Comment by srz1992** [comment]
@ota-meshi Ah, I had seen that rule but didn't notice the groups config so I thought it only applied to props. Thanks
